src/app/components/account/account-avatar/account-avatar.tsx

```diff
@@ -8,18 +8,38 @@ interface AccountAvatarProps extends BoxProps {
   name: string;
 }
 export const AccountAvatar = memo(({ publicKey, name, ...props }: AccountAvatarProps) => {
+  let firstGrapheme: string = '';
+  if (name?.includes('Account')) {
+    // Displays the account number.
+    //
+    // Assumes the name of the account is similar to "Account 4". This will
+    // probably break when the account has a BNS name with the string
+    // `"Account"` in it, although it's probably good enough for now.
+    firstGrapheme = name.split(' ')[1];
+  } else {
+    // Displays the first grapheme of the account name in uppercase.
+    //
+    // Using graphemes allows proper handling of emojis and other characters
+    // with multiple code points.
+    firstGrapheme = [...new Intl.Segmenter().segment(name)][0].segment;
+
+    // Will only transform text characters. E.g., emojis are unaffected.
+    firstGrapheme = firstGrapheme.toUpperCase();
+  }
+  const isFirstGraphemeEmoji = /\p{Extended_Pictographic}/u.test(firstGrapheme);
```

Comment:
This logic could be put into a function with a descriptive name like `getFirstGraphemeFromName()`.

Would also advocate for a more declarative (rather than imperative) style. Below example avoids redeclaration of a var, and omits the else statement

```ts
function getFirstGraphemeFromName(name: string) {
  // Displays the account number.
  //
  // Assumes the name of the account is similar to "Account 4". This will
  // probably break when the account has a BNS name with the string
  // `"Account"` in it, although it's probably good enough for now.
  if (name?.includes('Account')) return name.split(' ')[1];

  // Displays the first grapheme of the account name in uppercase.
  //
  // Using graphemes allows proper handling of emojis and other characters
  // with multiple code points.
  return [...new Intl.Segmenter().segment(name)][0].segment.toUpperCase();
}
```